cmd/kubeadm/app/cmd/options/certs.go

```diff
@@ -26,9 +26,11 @@ func AddCertificateDirFlag(fs *pflag.FlagSet, certsDir *string) {
 // AddCSRFlag adds the --csr-only flag to the given flagset
 func AddCSRFlag(fs *pflag.FlagSet, csr *bool) {
 	fs.BoolVar(csr, CSROnly, *csr, "Create CSRs instead of generating certificates")
+	fs.MarkDeprecated(CSROnly, "This flag will be removed in a future version. Please use kubeadm alpha certs generate-csr instead.")
```

Comment:
so i was going to recommend to include the deprecation message as part of the release note under "User facing change" in the PR description, but i realized the flags are used in a couple of places:
- kubeadm init phase certs
- kubeadm alpha certs renew

so the question here is - should we deprecate these flags in all places? (cc @fabriziopandini)

unlike "phase certs" where CSRs are less useful, for "renew" one might want to not renew right away and only generate the CSRs. i guess at this point we will be pointing users towards the new command that handles CSRs specifically and "renew" will be all about in-place renewal using the CA kubeadm looks for on disk? are my UX assumptions correct here?

in any case, we need to mention for which commands we are deprecating these flags in the release note.

Reply:
Ok. I've deprecated those flags for only `kubectl init phase certs ...` , 
not for `kubectl alpha certs renew --csr-only --csr-dir`.